core/src/main/scala/kafka/coordinator/group/CoordinatorPartitionWriter.scala

```diff
@@ -165,4 +166,21 @@ class CoordinatorPartitionWriter(
     // Required offset.
     partitionResult.lastOffset + 1
   }
+
+  override def deleteRecords(tp: TopicPartition, deleteBeforeOffset: Long, allowInternalTopicDeletion: Boolean): Unit = {
+    var deleteResults: Map[TopicPartition, DeleteRecordsPartitionResult] = Map.empty
+    replicaManager.deleteRecords(
+      timeout = 0L,
+      offsetPerPartition = Map(tp -> deleteBeforeOffset),
+      responseCallback = results => deleteResults = results,
+      allowInternalTopicDeletion
```

Comment:
I wonder if we could just set it to `true` here as this class is only used by the coordinator.